core/src/utils/overlays.ts

```diff
@@ -612,3 +612,45 @@ export const safeCall = (handler: any, arg?: any) => {
 };
 
 export const BACKDROP = 'backdrop';
+
+/**
+ * Constructs a trigger interaction for an overlay.
+ * Presents an overlay when the trigger is clicked.
+ *
+ * Usage:
+ * ```ts
+ * triggerController = createTriggerController();
+ * triggerController.configureTriggerInteraction(el, trigger);
+ * ```
+ */
+export const createTriggerController = () => {
+  let destroyTriggerInteraction: () => void;
+
+  const configureTriggerInteraction = (el: HTMLIonOverlayElement, trigger: string) => {
+    if (destroyTriggerInteraction) {
+      destroyTriggerInteraction();
+    }
+
+    const triggerEl = trigger !== undefined ? document.getElementById(trigger) : null;
+    if (!triggerEl) {
+      return;
+    }
+
+    const configureTriggerInteraction = (trigEl: HTMLElement, overlayEl: HTMLIonOverlayElement) => {
```

Comment:
The duplicate function name between this and the outer `configureTriggerInteraction` is a little confusing; could we add comments if this is intentional to work some magic, or else rename one of them?

Reply:
Agreed this is confusing. I renamed the public implementation to `initializeClickListener`.

Usage:
```ts
triggerController = createTriggerController();
triggerController.initializeClickListener(el, trigger);
```